File: ch21/C#/General/App_Code/SmartSessionPage2.cs
using System;
using System.Web;

public class SmartSessionPage2 : System.Web.UI.Page
{
    private const string MYPERSON = "myperson";

    public Person MyPerson
    {
        get
        {
            return (Person)Session[MYPERSON];
        }
        set
        {
            Session[MYPERSON] = value;
        }
    }
}

File: ch21/VB/General/App_Code/SmartSessionPage2.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Web

Public Class SmartSessionPage2
    Inherits System.Web.UI.Page

    Private Const MYSESSIONPERSONKEY As String = "myperson"

    Public Property MyPerson() As Person
        Get
            Return CType(Session(MYSESSIONPERSONKEY), Person)
        End Get
        Set(ByVal value As Person)
            Session(MYSESSIONPERSONKEY) = value
        End Set
    End Property
End Class
